include/zephyr/canbus/isotp.h

```diff
@@ -167,6 +179,16 @@ struct isotp_msg_id {
 	};
 	/** ISO-TP extended address (if used) */
 	uint8_t ext_addr;
+	/**
+	 * ISO-TP frame data length (TX_DL for TX address or RX_DL for RX address).
+	 *
+	 * Valid values are 8 for classical CAN or 8, 12, 16, 20, 24, 32, 48 and 64 for CAN-FD.
+	 * 0 will be interpreted as 8 for compatibility reasons.
```

Comment:
I was thinking that doing this at runtime could be useful 👍
Could send lower priority messages as smaller frames to reduce high priority latency.

Maybe also interpret 0 as 64 if CAN-FD is enabled, as that is probably the default expected behavior

Reply:
True.
As far as I remember, 8 is mandatory for classic CAN.

Reply:
Yes, I know 8 is mandatory for classical CAN, that's why values < 8 are not in that list. I would rather not ignore it for classical CAN, as you don't want people to choose 6 (or whatever) and implicitly get 8. Default 64 for CAN-FD will come with the next rebase once all the CI issues on main are fixed.